### DISCUSSION TITLE
How can I disable `pyright` and use `ruff-lsp` instead?

### DISCUSSION THREAD

**Discussion by neuromaancer** [LazyVim/LazyVim#1506] [Q&A] [answered]
I am using extra module of python, and I realized that the `pyright` is used by default for the static check. How can I disable it and use `ruff-lsp` instead? When I deleted the `pyright` from Mason, it will reinstall it as expected.

**Comment by dpetka2001** [comment]
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        pyright = {
          mason = false,
        },
      },
    },
  },
}
```
Put this in a file in `~/.config/nvim/lua/plugins/lsp.lua` (or whatever other name instead of `lsp.lua` you'd like), uninstall pyright from Mason UI and restart nvim. It shouldn't reinstall it.

**Reply by neuromaancer** [reply]
It doesn't work, I still can find the diagnostics from `pyright`, which is very strange, because I confirm that I have uninstalled it from Mason. `pyright` is not in Mason anymore, but I can still see the `pyright` diagnostics.

**Reply by dpetka2001** [reply]
For me it works fine. `:LspInfo` shows that only `ruff-lsp` is attached. So, whatever diagnostics I get come from `ruff-lsp`. What does `:LspInfo` show for you?

**Reply by neuromaancer** [reply]
Strange, it is still `pyright`, and this is what I put in the `lsp.lua`
```lua
return {
  "neovim/nvim-lspconfig",
  ---@class PluginLspOpts
  opts = {
    ---@type lspconfig.options
    servers = {
      pyright = {
        mason = false,
      },
      ruff_lsp = {},
      rust_analyzer = {},
    },
  },
}
```

**Reply by dpetka2001** [reply]
Please show what `:LspInfo` shows when opening a Python file.

**Reply by neuromaancer** [reply]
<img width="1111" alt="Screenshot 2023-09-28 at 11 58 58" src="https://github.com/LazyVim/LazyVim/assets/28112871/1bab4874-08cd-48b9-848b-45eefa832992">

<img width="451" alt="Screenshot 2023-09-28 at 12 00 53" src="https://github.com/LazyVim/LazyVim/assets/28112871/be1e26df-f8a7-4299-b728-82f651ca140b">

**Comment by dpetka2001** [comment]
Do you have pyright installed globally in your system?

**Reply by neuromaancer** [reply]
I don't think so, it might be `vscode` did it?

**Reply by dpetka2001** [reply]
If you take a look at the `cmd` section in `pyright`, you will see that the path is `/usr/local/bin/pyright-langserver --stdio`. That indicates you have installed it globally through some other way. Please trace back how you did that and uninstall it.

**Reply by dpetka2001** [reply]
What does `npm -g ls` show?

**Reply by neuromaancer** [reply]
Yeah, `pyright` is installed via `npm`, deleted it. But it always gives me warning that `pyright` is not found which is annoying.

**Reply by neuromaancer** [reply]
okay, just make the `autostart` to `false`, solved! Thanks!

**Reply by dpetka2001** [reply]
Yes, that's because it's still configured in `nvim-lspconfig` in the extras. The other thing you could do, is copy the contents of https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/extras/lang/python.lua to a file in `~/.config/nvim/lua/plugins/python.lua` and then make whatever modifications you want on your own, but in this case you should also remove the line from the `extras` in your `/lua/config/lazy.lua`.

**Reply by dpetka2001** [reply]
Cool, I didn't think about the `autostart`. Glad you solved it!

**Reply by dpetka2001** [reply]
@neuromaancer Maybe you should create a new answer with your config and mark it as solution, since my initial post doesn't cover what you want to do, so that others with the same problem will have an easier time finding the solution.

**Reply by neuromaancer** [reply]
Yes, just did it, thank you!

**Comment by neuromaancer** [comment, marked as answer]
Answer:
1. first, create a `lsp.lua` file in `plugins` folder.
2. second add the code below:
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        pyright = {
          mason = false,
          autostart = false,
        },
      },
    },
  },
}
```
3. remember if you install `pyright` global via `npm`, you need to uninstall globally by running `npm uninstall -g pyright`. 
4.  As suggested by @dpetka2001:
> The other thing you could do, is copy the contents of [main/lua/lazyvim/plugins/extras/lang/python.lua](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/extras/lang/python.lua) to a file in ~/.config/nvim/lua/plugins/python.lua and then make whatever modifications you want on your own, but in this case you should also remove the line from the extras in your /lua/config/lazy.lua.

**Comment by b1tburn3r20** [comment]
If anyone was still having trouble, they might have installed Pyright through Mason and forgot

You can verify by running `:Mason` and seeing if pyright is there

if Pyright is there, run `:MasonUninstall pyright`
